### ISSUE TITLE
[Feature request] Remove dependency on scikit-bio

### ISSUE THREAD

**Issue by jolespin** [fedarko/pyfastg#2]
`Scikit-bio` is pretty heavy weight for just calculating GC-content. I suggest that you remove this dependency to make it more accessible. 

e.g., 

```python
def GC(sequence:str, scaling_factor=100):
    sequence:str = sequence:str.upper()
    number_of_gc = sequence.count("G") + sequence.count("C")
    length_of_sequence = len(sequence)
    ratio_of_gc = number_of_gc/length_of_sequence 
    return ratio_of_gc * scaling_factor
```

**Comment by fedarko** [comment]
Thank you for noting this! It's been a while, but I believe scikit-bio actually has the additional extra use in this code of checking sequence validity:

https://github.com/fedarko/pyfastg/blob/3b99ba8624339e3efdc87de1d122e5f401b4537a/pyfastg/pyfastg.py#L142

^^ In addition to computing GC content, this line will fail if the sequence contains any characters that are not in the [IUPAC DNA alphabet](https://www.bioinformatics.org/sms/iupac.html)).

That being said, ~two years after writing this code I think this is overkill (plus this behavior might be undesirable if e.g. the sequences correspond to RNA, since sequences containing `U` will cause this line to fail).

So yes, 100% agreed -- we can drop the dependency, and change this line to compute GC content manually. (We should also remove the sentence in the README saying that "We assume that each node sequence (the line(s) between node declarations) consists only of valid DNA characters, as determined by [`skbio.DNA`](http://scikit-bio.org/docs/latest/generated/skbio.sequence.DNA.html).")

I'll put this on my TODO list, but I can't guarantee I'll get to it soon. If you or anyone else would like to take a stab at it, I'm happy to accept PRs. Thank you again for reporting this!

**Comment by jolespin** [comment]
Nice, yes I won't need this any time soon (plus the scikit-bio dependency isn't too unbearable! I have it in most my environments just a suggestion). Another question, can this handle rnaSPAdes output?

**Comment by fedarko** [comment]
Sounds good! I haven't tested this with rnaSPAdes output, but I imagine that the DNA validation thing mentioned above will cause problems until this issue is addressed. If rnaSPAdes' FASTG files are formatted similarly to ordinary SPAdes FASTG files, I think this should be compatible after addressing this.

**Comment by jolespin** [comment]
Biopython might be a good alternative that is more light weight https://biopython.org/wiki/Alphabet

**Comment by fedarko** [comment]
Just to follow up, this has been addressed! The latest pyfastg release (v0.1.0) no longer depends on scikit-bio; pyfastg's only remaining direct dependency is NetworkX.
